app/models/comment.rb

```diff
@@ -1,4 +1,7 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :post, counter_cache: true
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  def update_comments_count
```

Comment:
[OPTIONAL] It would be nice if you use `after_save` method for updating comments count and make `update_comments_count` private. More info: [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)